tests/test_sklearn_decision_tree_converters.py

```diff
@@ -142,6 +142,37 @@ def test_extra_trees_tree_trav_regressor_converter(self):
     def test_extra_trees_perf_tree_trav_regressor_converter(self):
         self._run_extra_trees_regressor_converter(1000, extra_config={"tree_implementation": "perf_tree_trav"})
 
+    # Used for DecisionTreeRegressor
+    def _run_decision_tree_regressor_converter(self, num_classes, extra_config={}):
+        # TODO: Extract method so that this can be reused across regressors.
```

Comment:
I believe it's about time we extract the method and let all the regressors here reuse it (Random forest, Extra trees and DT)

Reply:
Also, should we now change the class name from `TestSklearnRandomForestConverter` to `TestSklearnTreeConverter` maybe?

Reply:
These are good observations. Yes please go ahead with the refactoring.

Reply:
About the extracting the regressor methods: we can have some base method running all regression tests, but I like keeping the actual test functions separated because it is easier to understand which specific test is failing.

Reply:
Agreed. I was thinking we can extract this method:
```
warnings.filterwarnings("ignore")
        for max_depth in [1, 3, 8, 10, 12, None]:
            model = ... # this is now a input param
            X = np.random.rand(100, 200)
            X = np.array(X, dtype=np.float32)
            y = np.random.randint(num_classes, size=100)

            model.fit(X, y)
            pytorch_model = hummingbird.ml.convert(model, "pytorch", extra_config=extra_config)
            self.assertTrue(pytorch_model is not None)
            np.testing.assert_allclose(model.predict(X), pytorch_model.predict(X), rtol=1e-06, atol=1e-06)
```

And then each of the test methods would exist separately as it is now but calling the above method with the right params (model, etc).

Reply:
Addressed this. Can you take a look now?